### ISSUE TITLE
CMS Design: Homepage Hub List menu UI

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#12304]
## Description

As a Drupal admin, I want an interface that lets me create/edit a menu of items for the Hub List on the new homepage so that I can maintain it as Veteran needs evolve.

This design will enable implementation ticket #11968.

## Acceptance Criteria
- [x] Design for new "Add link view" exists in Figma (see [Jill's comment](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12304#issuecomment-1398907105) below for context)
- [x] UI includes default menu item options (see Jill's comment below); this includes among other items the Link and Link text
- [x] Menu also includes the following custom fields:
  - [x] "Icon" - select list that will not re-use any existing icon lists, and will include VA Dept Info; 
    - [x] help text: `Select an icon to appear next to this link's title.`
    - [x] list of icon/hub names exactly as in Benefit Hub Landing Page content type (save for the last item, `VA department info` which is unique to the new menu, and should not be present on BHLP). Note that the keys of these icons matter, since they are used on the FE. Therefore, the keys should match those from the BHLP. Engineer may choose a meaningful key for the additional 'VA department info' label.
        - None
        - Health care
        - Disability
        - Education
        - Records
        - Careers and employment
        - Pension
        - Housing
        - Life insurance
        - Burials and memorials
        - Family member benefits
        - Service member benefits
        - VA department info
  - [x] "Link summary" text
    - [x] Link summary helper text: `Displayed to Veteran under the link text`
- [x] Submit to CMS collab cycle team (Blake) for review
- [ ] Add Figma link to implementation ticket #11968 

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by jilladams** [comment]
Re: Hub list icon:  the discussion there was a longer term nice to have to make the 2 merged.

What's present on Benefit Hub Landing Page (BHLP) now is just a list, specified in that field. To make it usable across both that content type and Menu Items is Bigger: pull it out of BHLP, make it a taxonomy, make it useful across the two separate things, and figure out how to keep VA Dept Info out of the list that's present on the BHLP. Dave said: good for future, not necessary for now. Or at least: that's how I remember it. (@chri5tia / @dsasser does that line up with your take?)

**Comment by jilladams** [comment]
This is How Menus Work (TM). We don't need designs for every step -- just the last one, really, but setting up context for learning's sake. 

## 1. Access a specific menu
https://prod.cms.va.gov/admin/structure/menu displays a list of all existing Drupal menus
![Screen Shot 2023-01-20 at 12 23 18 PM](https://user-images.githubusercontent.com/85581471/213798708-ca013070-4e13-41e1-806a-9c4f37a4619b.png)

## 2. View a menu
e.g. https://prod.cms.va.gov/admin/structure/menu/manage/header-megamenu
* Menus contain Menu Items. This view allows users to manage:
  * Menu name 
  * Administrative summary 
  * Order of Menu Items
  * Adding new links
* Menu Items are displayed in weighted order
* Sighted users can drag / drop order
* You can click `Show row weights` to get to an interface where you can adjust menu order without drag/drop

### 2a. Drag/drop view
![Screen Shot 2023-01-20 at 12 21 53 PM](https://user-images.githubusercontent.com/85581471/213799055-9dbaf5e8-e102-44f6-958c-7374f504739a.png)

### 2b. Row weights view
![Screen Shot 2023-01-20 at 12 21 58 PM](https://user-images.githubusercontent.com/85581471/213799106-87842d81-a0f2-4947-b695-6003afd015ff.png)

## 3. Add Link view
**This is the view we are designing for the new Hub List menu.** Adding a Link is configurable per-menu, to a certain extent. Some default fields must appear here, and we can add new fields as we need. 

### Default fields
* Menu link title (aka Hub Label, for this project)
* Link
* Enabled
* View Mode
* Description
* Attributes
* Parent link
* Weight

### Custom fields
* Hub icon - select list
* Hub description - paragraph text

### Header megamenu example
e.g. https://prod.cms.va.gov/admin/structure/menu/manage/header-megamenu/add?destination=/admin/structure/menu/manage/header-megamenu --- example of adding a Menu Item to the Header megamenu. Shows the custom field "Related Promo."

![header megamenu menu item](https://user-images.githubusercontent.com/85581471/213799795-f957aa34-ee83-48a5-a7cb-ddd648803f87.png)

### VA.gov footer bottom rail example
https://prod.cms.va.gov/admin/structure/menu/item/11510/edit?destination=/admin/structure/menu/manage/footer-bottom-rail -- example adding a Menu Item to the VA.gov Footer bottom rail menu

![footer bottom rail](https://user-images.githubusercontent.com/85581471/213799826-7db4989a-fbde-4bd7-bad6-adf46631b419.png)

**Comment by wesrowe** [comment]
Per [slack thread](https://dsva.slack.com/archives/CT4GZBM8F/p1674252926815819), I'm going to rename the field that holds text for homepage UI to "Link summary" (previous: Link description).

**Comment by dsasser** [comment]
@wesrowe this AC seems a bit confusing to me, as it suggests the icon list is the same as those on the BHLP, yet the one we are adding for the 12th hub item is not present on BHLP:

![Screen Shot 2023-01-24 at 11 45 45 AM](https://user-images.githubusercontent.com/221539/214393763-e90b04f7-7693-4f8b-9ad9-d45203e9ed61.png)

Could we reword this to address the difference between both icon sets?

**Comment by thejordanwood** [comment]
@BlakeOrgan @EWashb Here are [the Figma comps](https://www.figma.com/file/nPSOXFJN41CZGmK2eXgvoy/HUB-List-Menu?node-id=0%3A1&t=T2hFX7UrqIdDorPF-1) for the homepage hub list menu changes. Please let me know if there are any changes that I need to make!

cc: @jilladams @wesrowe

**Comment by jilladams** [comment]
Adding @BlakeOrgan as assignee just to reflect that his feedback is next step, toward end of this sprint.

**Comment by jilladams** [comment]
From refinement: we would like to hide the Attributes accordion from this design. For this particular menu, and its use, we don't want editors to be able to set Class or Target, so we'll prevent them from showing.

**Comment by thejordanwood** [comment]
@jilladams I removed the attributes accordion from the mockups.

**Comment by BlakeOrgan** [comment]
Had a meeting with @thejordanwood and @swirtSJW to discuss design edits. Comments are in Figma.

**Comment by thejordanwood** [comment]
Design edits have been made and we're ready to move forward with this. Design can be found in [this Figma file](https://www.figma.com/file/nPSOXFJN41CZGmK2eXgvoy/HUB-List-Menu?node-id=46%3A1546&t=q4uq7AsNpchdE1FZ-1). Closing this ticket so we can move onto [Drupal implementation](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11968).

cc: @BlakeOrgan @jilladams @wesrowe

**Comment by jilladams** [comment]
Follow up implementation note from Drupal eng: if a field is required, "None" should not be an option in the list. We need Hub Icon to be required in this menu, so will update the design to remove the "None" option. 

This differs from how Benefit Hub Landing Pages currently behave. There, the field is optional. However: there are no hubs without an Icon, and there shouldn't be for the sake of the homepage treatment. So we are changing behavior here.
